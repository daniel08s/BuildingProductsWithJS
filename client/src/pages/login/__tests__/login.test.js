/* global test expect shallow React */
// npm packages
import configureMockStore from 'redux-mock-store';

// our packages
import LoginPage, {Login} from '../index';

// create mock store
const mockStore = configureMockStore();

const user = {
  id: '1',
  login: 'testuser',
  registrationDate: new Date(2018, 1, 1, 1, 1, 1, 1),
};

const token = '0';

test('# LoginPage Wrapper ', () => {
  const store = mockStore({auth: {token}});

  const wrapper = shallow(<LoginPage store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Login page', () => {
  const newUser = {
    login: 'testuser',
    password: '123',
    remember: true,
  };
  const onLoginClick = ({login, password, remember}) => {
    expect(login).toBe(newUser.login);
    expect(password).toBe(newUser.password);
    expect(remember).toBeTruthy();
  };
  const navToHome = () => expect(true).toBeTruthy();
  const component = (
    <Login
      token={token}
      onLoginClick={onLoginClick}
      navToHome={navToHome}
    />
  );

  // test rendering
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();

  // mount for testing
  const app = mount(component);

  // test navHome
  app.find('#inputUsername').getDOMNode().value = newUser.login;
  app.find('#inputPassword').getDOMNode().value = newUser.password;
  app.find('#inputRemember').getDOMNode().checked = newUser.remember;

  // click button
  app.find('button').simulate('click');
});
