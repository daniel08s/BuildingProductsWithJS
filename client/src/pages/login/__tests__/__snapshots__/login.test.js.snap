// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Login page 1`] = `
<form
  className="formSignin"
>
  <h1
    className="h1 mb-1 font-weight-normal"
  >
    Experts portal
  </h1>
  <p
    className="h3 mb-3 font-weight-normal"
  >
    Please sign in
  </p>
  <label
    className="sr-only"
    htmlFor="inputUsername"
  >
    Username
  </label>
  <input
    className="form-control"
    id="inputUsername"
    placeholder="Username"
    required={true}
    type="text"
  />
  <label
    className="sr-only"
    htmlFor="inputPassword"
  >
    Password
  </label>
  <input
    className="form-control"
    id="inputPassword"
    placeholder="Password"
    required={true}
    type="password"
  />
  <div
    className="checkbox mb-3"
  >
    <label
      htmlFor="inputRemember"
    >
      <input
        id="inputRemember"
        type="checkbox"
      />
       Remember me
    </label>
  </div>
  <button
    className="btn btn-lg btn-primary btn-block"
    onClick={[Function]}
    type="submit"
  >
    Login
  </button>
  <p
    className="h5 font-weight-normal"
  >
    Not registered? 
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/register"
    >
      Click here.
    </Link>
  </p>
  <p
    className="mt-5 mb-3 text-muted"
  >
    Â© 2017-2018
  </p>
</form>
`;

exports[`# LoginPage Wrapper  1`] = `
<Login
  navToHome={[Function]}
  onLoginClick={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  token="0"
/>
`;
